src/main/java/service/operation/SupplyOperation.java

```diff
@@ -0,0 +1,23 @@
+package service.operation;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+
+public class SupplyOperation implements OperationHandler {
+    private FruitDao fruitDao;
+
+    public SupplyOperation() {
+        this.fruitDao = new FruitDaoImpl();
+    }
+
+    @Override
+    public void handle(String fruit, int quantity) {
+        int balanceAfterSupply = fruitDao.getFruitBalance(fruit) + quantity;
+        if (balanceAfterSupply >= 0) {
```

Comment:
The check for `balanceAfterSupply >= 0` is redundant in the `SupplyOperation` as the supply operation should always increase the balance. You can remove this check.